### PR TITLE
Fix duplicate contributor in members query

### PR DESCRIPTION
**Pull request by willm30**
## Description

dupl

### CODE REVIEW

File: amplify/backend/function/getMembersForColony/src/index.js [hunk 1]
```diff
@@ -121,177 +121,142 @@ exports.handler = async (event) => {
...
-      watchers.push({
-        address: item.user.id,
-        user: item.user,
-      });
-    }
-  });
-
-  // There will be no Watchers outside of the root domain
-  if (domainId > Id.RootDomain) {
-    watchers.length = 0;
```
Comment (willm30) [inline amplify/backend/function/getMembersForColony/src/index.js:162]: This is just weird.

File: amplify/backend/function/getMembersForColony/src/index.js [hunk 2]
```diff
@@ -121,177 +121,142 @@ exports.handler = async (event) => {
...
+        address: user.id,
+        user,
+      })),
     };
   }
 
-  // Identify watchers & contributors
-  data?.getColonyByAddress?.items[0]?.watchers.items.forEach((item) => {
-    if (
-      addressesWithReputation?.some(
```
Comment (willm30) [inline amplify/backend/function/getMembersForColony/src/index.js:144]: This double loop is slower than using a hash set. m * n versus m + n

File: amplify/backend/function/getMembersForColony/src/index.js [hunk 3]
```diff
@@ -121,177 +121,142 @@ exports.handler = async (event) => {
...
+    // Or it's a subdomain, in which case we can return two empty arrays since you can't watch a subdomain
+    return {
+      contributors: [],
+      watchers: [],
+    };
   }
 
-  // get reputation for each address
-  const contributorsWithReputation = await Promise.all(
-    contributors.map(async (contributor) => {
```
Comment (willm30) [inline amplify/backend/function/getMembersForColony/src/index.js:196]: This logic was hard to follow, but basically it handles calculating the user's reputation in the domain. I have simplified.

File: amplify/backend/function/getMembersForColony/src/index.js [hunk 4]
```diff
@@ -121,177 +121,142 @@ exports.handler = async (event) => {
...
-  });
-
-  // There will be no Watchers outside of the root domain
-  if (domainId > Id.RootDomain) {
-    watchers.length = 0;
-  }
-
-  // now catch addresses that have reputation but are not in the watchers list
-  // i.e. address that was awarded reputation but has not joined the colony
-  if (addressesWithReputation?.length !== contributors?.length) {
```
Comment (willm30) [inline amplify/backend/function/getMembersForColony/src/index.js:167]: I presume this is responsible for my duplicate contributor, probably combined with some specific user flow like being awarded rep while not joined to a colony and then joining... or something like that. It's hard to tell. 

I have changed so that we generate contributors straight from the source of truth provided by the chain, instead of trying to splice together two separate lists. This should solve the problem.

File: amplify/backend/function/getMembersForColony/src/index.js [hunk 5]
```diff
@@ -68,209 +67,196 @@ exports.handler = async (event) => {
...
+
   if (errors || !data) {
     const [error] = errors;
     throw new Error(
       error?.message || 'Could not fetch colony data from DynamoDB',
     );
   }
 
-  // Incase there are members but no reputation on the colony,
-  // we can stop here and return the list of watchers
```
Comment (willm30) [inline amplify/backend/function/getMembersForColony/src/index.js:125]: Contributors can also have no rep if they have permissions, therefore we can't terminate early here

**Review by jakubcolony — APPROVED** [review]
Looks good, nice fix! 💪

**Review by ArmandoGraterol — APPROVED** [review]
Excellent work! 

Do we have an issue to fix the members not displaying the permission labels in the list? This is how the list looks in the Dapp in prod:
FireShot Capture 121 - Members - Colony - Metacolony - xdai colony io

Also, toggling the permission sorter doesn't do anything for me

**Comment by willm30** [comment]
@ArmandoGraterol Yep, keeping track of both things you mention in the QA project 👌
